File: Models/Comment.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace OsloMetAngular.Models
{
    // Data model (data schema) for DB
    // User and Post are navigation properties.
    
    public class Comment
    {
        [JsonProperty("CommentID")]
        public int CommentID { get; set; }
        [JsonProperty("CommentText")]
        [StringLength(200)]
        public string CommentText { get; set; } = string.Empty;
        [JsonProperty("PostDate")]
        public string? PostDate { get; set; }
        [JsonProperty("UserId")]
        public int UserId { get; set; } = 1;
        //[Newtonsoft.Json.JsonIgnore]
        public virtual User User { get; set; } = default!;
        [JsonProperty("PostID")]
        public int PostID { get; set; }
        //[Newtonsoft.Json.JsonIgnore]
        public virtual Post Post { get; set; } = default!;


    }
}

File: ClientApp/src/app/posts/comment.ts
import { IUser, User } from "../user/user";
import { IPost, Post } from "./post";

export interface IComment {
  CommentID: number;
  CommentText: string;
  PostId: number;
  UserId: number;
  PostDate: string;
  user: IUser;
  post: IPost;
}
export class Comment implements IComment {
    CommentID: number = 0;
    CommentText: string = "";
    PostId: number = 0;
    UserId: number = 0;
    PostDate: string = "";
    user: IUser = new User();
    post: IPost = new Post();
}
